src/Cosserat/mapping/BaseCosserat.inl

```diff
@@ -1,274 +1,250 @@
+  // d_curv_abs_section;
+  helper::ReadAccessor<Data<type::vector<double>>> curv_abs_frames =
+      d_curv_abs_frames;
+
+  m_framesExponentialSE3Vectors.clear();
+  m_nodesExponentialSE3Vectors.clear();
+  m_nodesLogarithmeSE3Vectors.clear();
+  const unsigned int sz = curv_abs_frames.size();
+
+  // Compute exponential at each frame point
+  for (size_t i = 0; i < sz; i++) {
+    Transform g_X_frame_i;
+
+    const Coord1 strain_n = inDeform[m_indicesVectors[i] -
+                                     1]; // Cosserat reduce coordinates (strain)
+                                         // the size varies from 1 to 6
+    //  The distance between the frame and the closest beam node toward the base
+    const SReal curv_abs_x =
+        m_framesLengthVectors[i]; // curv_abs_x = frame_curv_abs - L_(n-1)
+    computeExponentialSE3(curv_abs_x, strain_n, g_X_frame_i);
+    m_framesExponentialSE3Vectors.push_back(g_X_frame_i);
+
+    if (d_debug.getValue()) {
+      msg_info("BaseCosserat:")
+          << "_________________" << i << "_________________________";
+      msg_info("BaseCosserat:")
+          << "x :" << curv_abs_x << "; strain :" << strain_n;
+      msg_info("BaseCosserat:")
+          << "m_framesExponentialSE3Vectors :" << g_X_frame_i;
     }
-
-    //Compute the exponential at the nodes
-
-    m_nodesExponentialSE3Vectors.push_back(Transform(type::Vec3(0.0,0.0,0.0),type::Quat(0.,0.,0.,1.)));
-
-    for (unsigned int  j = 0; j < inDeform.size(); j++) {
-        type::Vec3 k = inDeform[j];
-        double  x = m_BeamLengthVectors[j];
-        Transform T; computeExponentialSE3(x,k,T) ;
-        m_nodesExponentialSE3Vectors.push_back(T);
-
-        if(d_debug.getValue()){
-            msg_info("BaseCosserat:")<< "_________________Beam Node Expo___________________"<<j;
-            msg_info("BaseCosserat:")<< "Node m_framesExponentialSE3Vectors :"<< T;
-            msg_info("BaseCosserat:")<< "_________________Beam Node Expo___________________";
-        }
-        //////////////////
-        //        Eigen::Matrix4d gX = convertTransformToMatrix4x4(T);
-        //        Eigen::Matrix4d log_gX= (1.0/x) * computeLogarithm(x, gX);
-        //        std::cout << "k : \n"<< k << std::endl;
-        //        std::cout << "The logarithm : \n"<< log_gX << std::endl;
-        //        m_nodesLogarithmSE3Vectors.push_back(log_gX);
+  }
+
+  // Compute the exponential on the nodes
+  m_nodesExponentialSE3Vectors.push_back(
+      Transform(type::Vec3(0.0, 0.0, 0.0),
+                type::Quat(0., 0., 0., 1.))); // The first node.
+
+  for (unsigned int j = 0; j < inDeform.size(); j++) {
+    Coord1 strain_n = inDeform[j]; // Strain_n
+    const SReal curv_abs_x =
+        m_BeamLengthVectors[j]; // curv_abs_x = L_n - L_(n-1)
+
+    Transform g_X_node_j;
+    computeExponentialSE3(curv_abs_x, strain_n, g_X_node_j);
+    m_nodesExponentialSE3Vectors.push_back(g_X_node_j);
+
+    if (d_debug.getValue()) {
```

Comment:
See suggestion about using single message, not string identifier and not using d_debug.